django/db/migrations/migration.py

```diff
@@ -122,6 +123,32 @@ def apply(self, project_state, schema_editor, collect_sql=False):
             else:
                 # Normal behaviour
                 operation.database_forwards(self.app_label, schema_editor, old_state, project_state)
+            return (old_state, project_state)
+
+        def _apply_operations(operations):
+            if not operations:
+                return
+
+            injected_operations = []
+            for operation in operations:
+                from_state, to_state = _apply_operation(operation)
+                emit_post_operation_signal(
+                    migration=self,
+                    operation=operation,
+                    from_state=from_state,
+                    to_state=to_state,
+                    injected_operations=injected_operations,
+                )
+
+            _apply_operations(injected_operations)
```

Comment:
@charettes What do you think about executing the injected operations inside the loop?

Take the following migration:
```
- RenameModel
- CreateModel
- RunPython(do_something_with_permissions_and_contenttypes)
```

Current implementation (LRN):
```
- RenameModel
- CreateModel
- RunPython(do_something_with_permissions_and_contenttypes)
- RenameContentType
- CreateContentType
- RenamePermissions
- CreatePermissions
```

Suggestion (LNR):
```
- RenameModel
- RenameContentType
- RenamePermissions
- CreateModel
- CreateContentType
- CreatePermissions
- RunPython(do_something_with_permissions_and_contenttypes)
```